provider/provider_yaml_test.go

```diff
@@ -211,6 +209,21 @@ func TestNoGlobalProjectWarning(t *testing.T) {
 	)
 }
 
+func TestGlobalProjectNoProjectWarning(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("Skipping in testing.Short() mode, assuming this is a CI run without GCP creds")
+	}
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+	proj := getProject()
+	test := pulumitest.NewPulumiTest(t, "test-programs/project-bucket",
+		opttest.LocalProviderPath(providerName, filepath.Join(cwd, "..", "bin")))
+
+	test.SetConfig(t, "gcp:disableGlobalProjectWarning", "true")
+	test.SetConfig(t, "gcpProj", proj)
+	test.Up(t)
```

Comment:
You need to capture the stdout from the result of `test.Up(t)` here and assert that it does not contain the warning.